test/distributed/_shard/sharded_tensor/ops/test_linear.py

```diff
@@ -125,22 +136,26 @@ def _run_sharded_linear(
         optim.step()
         self.assertNotEqual(previous, local_linear.weight)
         previous_sharded_weight = sharded_weight.clone()
-        previous_sharded_bias = sharded_linear.bias.clone()
+        previous_sharded_bias = sharded_linear.module.bias.clone()
+
+        # Use different learning rate (scale by world_size) for bias since it is replicated via DDP.
+        bias_optim = torch.optim.SGD([sharded_linear.module.bias], lr=0.1 * self.world_size)
```

Comment:
why we need to use different learning rate instead of a single one? Im not quite sure I can get it, is this also what the end user code look like?

Reply:
This is covered in our DDP docs: https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel

```
When a model is trained on M nodes with batch=N, the gradient will be M times smaller when compared to the same model trained on a single node with batch=M*N if the loss is summed (NOT averaged as usual) across instances in a batch (because the gradients between different nodes are averaged). You should take this into consideration when you want to obtain a mathematically equivalent training process compared to the local training counterpart. But in most cases, you can just treat a DistributedDataParallel wrapped model, a DataParallel wrapped model and an ordinary model on a single GPU as the same (E.g. using the same learning rate for equivalent batch size).
```

Reply:
I see, does it mean that we are only doing this (initializing two optimizers) purely for local megatron numerical parity, but for most cases, we don't need to do it with DDP?

Reply:
Numerically, they are the same. But isn't this going to confuse users because we need to use two learning rates for bias and weight. I am not sure if using separate learning rate is a common practice or not.

Reply:
I think we will probably have to document this similar to how DDP above documents how to use an appropriate learning rate.

Reply:
I guess what I want to say here is that, this might still cause confusions to users even if we document it. Hopefully, if we can explain it well in the document so that users won't have too many questions.